feature/google/src/main/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -625,6 +626,9 @@ class BillingWrapper(
 
             val responseCode =
                 if (purchases == null && billingResult.responseCode == BillingClient.BillingResponseCode.OK) {
+                    // Result being ok but with a Null purchase used to happen when doing a product change
+                    // in DEFERRED proration mode in Billing Client <= 4. Should not happen in Billing Client 5+ since
+                    // we get the transaction for the previous product.
```

Comment:
Could we add some more details to the error message in this case... It would be good to get this information clearly in case we receive any reports about it not working as expected. Right now, the error will just be a normal `StoreProblemError`, which might make it difficult to track the cause.

Reply:
Added a special error message. This should give us/the developer enough context to figure out the error. Was hesitant to move the whole comment inside the message but seemed too much.